PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalCreditFinancing.kt

```diff
@@ -0,0 +1,67 @@
+package com.braintreepayments.api.paypal
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Represents the PayPal credit financing response.
+ *
+ * @property isCardAmountImmutable Indicates whether the card amount is editable after payer's
+ * acceptance on PayPal side.
+ * @property monthlyPayment Estimated amount per month that the customer will need to pay including
+ * fees and interest.
+ * @property payerAcceptance Status of whether the customer ultimately was approved for and chose to
+ * make the payment using the approved installment credit.
+ * @property totalCost Estimated total payment amount including interest and fees the user will pay
+ * during the lifetime of the loan.
+ * @property totalInterest Estimated interest or fees amount the payer will have to pay during the
+ * lifetime of the loan.
+ * @property term Length of financing terms in months.
+ */
+@Parcelize
+data class PayPalCreditFinancing internal constructor(
+    val isCardAmountImmutable: Boolean = false,
+    val monthlyPayment: PayPalCreditFinancingAmount? = null,
+    val payerAcceptance: Boolean = false,
```

Comment:
```suggestion
    val hasPayerAcceptance: Boolean = false,
```
🧶 : Any opposition to keeping the `has-` prefix? I really like this portion of the [Swift naming guidelines](https://www.swift.org/documentation/api-design-guidelines/#boolean-assertions):

> Uses of Boolean methods and properties should read as assertions about the receiver when the use is nonmutating, e.g. x.isEmpty, line1.intersects(line2).

I understand it might be extra work if the compiler generates `getHasPayerAcceptance()` though. In that case, we'd need to override the generated property accessor with a custom `get set` declaration.

Reply:
Good callout. I tried to find a solution that would scale for all of the booleans we have in our request and response objects.

```kotlin
class Test(
    @JvmField // removes the generated java getter function
    val hasPayerAcceptance: Boolean = false,
) {
    
    fun hasPayerAcceptance() = hasPayerAcceptance
}
```

One downside to this is that in Kotlin there will be 2 ways to get the boolean value:
```kotlin
val test = Test(true)
test.hasPayerAcceptance()
test.hasPayerAcceptance
```

This pattern also will not work for abstract classes containing booleans, as the `@JvmStatic` annotation requires the property to be final.

Would we be okay with the generated `getHasPayerAcceptance()` for java integrations? or would we want to keep the naming as `payerAcceptance`?

I'm leaning towards just exposing `getHasPayerAcceptance()` given the adoption of Kotlin and that devs would still see the correct function in IDE autocomplete.

Reply:
I see yeah I figured maintainability would be a concern. I'd lean towards `getHasPayerAcceptance()` as the past of least resistance. Another option could be to refactor the property to something like `isPayerAccepted`, but i'm being super nit picky. I trust your judgement this is a take-or-leave for me.